pd client panic

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x35b6960]

goroutine 540 [running]:
github.com/tikv/pd/client.(*client).ScanRegions(0xc003828000, {0xcab5750, 0xc00459dae0}, {0xc004310ab0, 0x12, 0x12}, {0xc004310ac8, 0x12, 0x12}, 0x400, ...)
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20240103101103-a4d2f1ca365a/client.go:1040 +0x8c0
github.com/tikv/client-go/v2/internal/locate.(*RegionCache).scanRegions(0xc0019fe000, 0xc003e37dd0, {0xc004310ab0, 0x12, 0x12}, {0xc004310ac8, 0x12, 0x12}, 0x400)
	/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231227070846-61c486af13a5/internal/locate/region_cache.go:1864 +0x54a
github.com/tikv/client-go/v2/internal/locate.(*RegionCache).BatchLoadRegionsWithKeyRange(0xc0019fe000, 0x23?, {0xc004310ab0, 0x12, 0x12}, {0xc004310ac8, 0x12, 0x12}, 0x0?)
	/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231227070846-61c486af13a5/internal/locate/region_cache.go:1425 +0xd7
github.com/pingcap/tiflow/cdc/kv.(*SharedClient).divideAndScheduleRegions(0xc0013c2140, {0xcab5750?, 0xc00459dae0}, {0xfffffffffffffffd, {0xc004310ab0, 0x12, 0x12}, {0xc004310ac8, 0x12, 0x12}}, ...)
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/kv/shared_client.go:465 +0x737
github.com/pingcap/tiflow/cdc/kv.(*SharedClient).handleRequestRanges.func1()
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/kv/shared_client.go:438 +0xd5
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/pkg/mod/golang.org/x/sync@v0.5.0/errgroup/errgroup.go:75 +0x98
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1080
	/go/pkg/mod/golang.org/x/sync@v0.5.0/errgroup/errgroup.go:72 +0x125

```

--- a/go.mod
+++ b/go.mod
@@ -90,7 +90,7 @@ require (
 	github.com/thanhpk/randstr v1.0.6
 	github.com/tikv/client-go/v2 v2.0.8-0.20231227070846-61c486af13a5
 	github.com/tikv/pd v1.1.0-beta.0.20231212061647-ab97b9a267f3
-	github.com/tikv/pd/client v0.0.0-20240103101103-a4d2f1ca365a
+	github.com/tikv/pd/client v0.0.0-20240108040922-c94810e110ce
 	github.com/tinylib/msgp v1.1.6
 	github.com/uber-go/atomic v1.4.0
 	github.com/vmihailenco/msgpack/v5 v5.3.5

